function/expression.go

```diff
@@ -80,6 +80,17 @@ func (e EvaluationContext) WithTimerange(t api.Timerange) EvaluationContext {
 	return e
 }
 
+// WithPredicate duplicates the EvaluationContext but with a new predicate conjunct to it.
+func (e EvaluationContext) WithPredicate(p predicate.Predicate) EvaluationContext {
+	e.Predicate = predicate.AndPredicate{
```

Comment:
Use `func predicate.All(Predicate...) Predicate` instead of an explicit `AndPredicate`- it hands cases like `nil` predicates better